ibis/backends/clickhouse/__init__.py

```diff
@@ -260,10 +259,21 @@ def _normalize_external_tables(self, external_tables=None):
                 raise TypeError(f'Schema is empty for external table {name}')
 
             df = obj.data.to_frame()
-            structure = list(zip(schema.names, map(serialize, schema.types)))
-            external_tables_list.append(
-                dict(name=name, data=df.to_dict("records"), structure=structure)
+            structure = list(
+                zip(
+                    schema.names,
+                    map(
+                        serialize,
+                        (
+                            # unwrap nested structures
+                            typ.copy(nullable=not typ.is_nested())
```

Comment:
Can you add a bit more context why is it needed here?

Reply:
Yep. Will do.

Reply:
Done.